### PR TITLE
Runtime: Handle ANSI escape sequences with ":" without crashing

### PR DESCRIPTION
**Pull request by kovasap**

### CODE REVIEW

**Comment by blueyed** [comment]
Thanks!

Can you describe the issue please?
Also a test for this would be great, but first it should be clear how to reproduce this.

**Comment by kovasap** [comment]
Yes sorry.  The following command should reproduce the issue:

nvim -c 'set ft=man' test_ansi.txt 

[test_ansi.txt](https://github.com/neovim/neovim/files/3303627/test_ansi.txt)

I noticed this when using neovim to read the contents of my terminal as a part of my workflow.  I think the colons help describe background colors for characters, but am not sure exactly.  This patch was mostly to avoid errors that resulted in neovim just giving up on interpreting the color codes and having them in my final buffer, making it unreadable.

**Comment by justinmk** [comment]
Thanks for the effort! This code is tested in `test/functional/plugin/man_spec.lua`. Could you try adding a test case for this problem?

You can [run the test](https://github.com/neovim/neovim/blob/master/test/README.md#running-tests) like this:

    TEST_FILE=test/functional/plugin/man_spec.lua make functionaltest

**Comment by kovasap** [comment]
Ok thanks!  I think i'm close with getting the test to pass, but am a little confused about the last line of output I'm getting (and the meaning of the | characters - they seem to move around for me when I change them).  Any tips?

**Comment by justinmk** [comment]
> Ok thanks! I think i'm close with getting the test to pass, but am a little confused about the last line of output I'm getting (and the meaning of the | characters - they seem to move around for me when I change them). Any tips?

Most likely just whitespace needs to be adjusted. The `|` are just visual aids for you, they are not part of the input/output.

```
[1m[31m[  ERROR   ][0m[0m [36m...ld/neovim/neovim/test/functional/plugin/man_spec.lua[0m @ [36m135[0m: [1m:Man man.lua: highlight_line() handles : characters in input[0m
test/functional/ui/screen.lua:554: Row 2 did not match.
Expected:
  |^testing                                             |
  |*    0      1      2      3                         |
  |    4      5      6      7      8      9           |
  |   10     11     12     13     14     15           |
  |   16                                                  |
Actual:
  |^testing                                             |
  |*    0      1      2      3                          |
  |    4      5      6      7      8      9            |
  |   10     11     12     13     14     15            |
  |{b:-- INSERT --}                                        |
```

The line marked with `*` shows the first mismatched line. The "actual" case has one more space character than the "expected" case.

**Comment by kovasap** [comment]
Oh ok cool!  What is the INSERT thing at the bottom?

**Comment by blueyed** [comment]
> What is the INSERT thing at the bottom?

`:set showmode` :)

**Comment by kovasap** [comment]
Ok, I think I almost have it working.  One more question: I'm a bit stuck/curious about why the "16" on the last line of my test does not show up.  When I open my test file with man mode it does show up, but it doesn't in the test.  Is there anything else about the test environment that is different than just regular nvim?

**Comment by justinmk** [comment]
> Ok, I think I almost have it working. One more question: I'm a bit stuck/curious about why the "16" on the last line of my test does not show up. When I open my test file with man mode it does show up, but it doesn't in the test. Is there anything else about the test environment that is different than just regular nvim?

The nvim instance for each test case starts with these settings:

https://github.com/neovim/neovim/blob/93f8c2793cf5020d6fca00b6f10e4a158578034f/test/functional/helpers.lua#L29-L31

**Comment by kovasap** [comment]
I think I got it!  Thanks for all the help.

**Comment by justinmk** [comment]
Merged, thanks @kovasap !

(squashed, fixed lint errors)

File: test/functional/plugin/man_spec.lua [hunk 1]
```diff
@@ -131,5 +131,32 @@ describe(':Man', function()
...
+      -- what input looks like on terminal
+      -- [30m    0  [31m    1  [32m    2  [33m    3  
+      -- [34m    4  [35m    5  [36m    6  [37m    7  [90m    8  [91m    9  
+      -- [92m   10  [93m   11  [94m   12  [95m   13  [96m   14  [97m   15  
+      -- [38:5:16m   16  
+
+      -- [40m    0  [41m    1  [42m    2  [43m    3  
+      -- [44m    4  [45m    5  [46m    6  [47m    7  [100m    8  [101m    9  
+      -- [102m   10  [103m   11  [104m   12  [105m   13  [106m   14  [107m   15  
+      -- [48:5:16m   16
```
Comment (justinmk) [inline test/functional/plugin/man_spec.lua]: This is a good type of comment, but in this case it's redundant with the `rawfeed()` call so I removed it.

**Pull request closed without merging**